src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -1,47 +1,151 @@
 package core.basesyntax;
 
 import java.util.List;
+import java.util.Objects;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+    private int size;
+    private Node<T> first;
+    private Node<T> last;
+
     @Override
     public void add(T value) {
+        if (size == 0) {
+            first = last = new Node<>(null, value, null);
+
+        } else {
+            Node<T> newNode = new Node<>(last, value, null);
+            last.next = newNode;
+            last = newNode;
+        }
+        size++;
     }
 
     @Override
     public void add(T value, int index) {
+        if (index < 0 || index > size) {
+            throw new IndexOutOfBoundsException("Index " + index + " out off list length " + size);
+        }
+        if (index == size) {
+            add(value);
+            return;
+        }
+        Node<T> currentNode = getNodeByIndex(index);
+        Node<T> newNode = new Node<>(currentNode.prev, value, currentNode);
+        if (index == 0) {
+            first = newNode;
+        } else {
+            currentNode.prev.next = newNode;
+        }
+        currentNode.prev = newNode;
+        size++;
     }
 
     @Override
     public void addAll(List<T> list) {
+        for (T element : list) {
+            add(element);
+        }
     }
 
     @Override
     public T get(int index) {
-        return null;
+        checkIndex(index);
+        Node<T> searchElement = first;
+        for (int i = 0; i < size; i++) {
+            if (i == index) {
+                break;
+            }
+            searchElement = searchElement.next;
+        }
+        return searchElement.item;
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (index >= size || index < 0) {
+            throw new IndexOutOfBoundsException("Index " + index + " out off list length " + size);
+        }
```

Comment:
can we replace this part of code with method `checkIndex()`)

Reply:
I don't understand what's wrong